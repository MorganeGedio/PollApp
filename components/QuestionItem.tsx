import * as React from "react";
import { View, Text, Image, StyleSheet, TouchableOpacity } from "react-native";

export default function QuestionItem(props) {
  return (
    <TouchableOpacity onPress={props.navigateToQuestion}>
      <View style={styles.container}>
        <View style={styles.baseText}>
          <Text style={styles.titleText}>{props.title}</Text>
          <Text style={styles.descriptionText}>Created on: {props.date}</Text>
          <Image style={styles.icon} source={require('../assets/images/point-to.png')} />
        </View>
        <View style={styles.box}>
          <Image style={styles.image} source={require('../assets/images/hand.png')} />
        </View>
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    marginLeft: 30,
    marginRight: 30,
    marginTop: 30,
    height: 120,
    flexDirection: "row",
    backgroundColor: "#dfcdc3",
    justifyContent: "space-between",
    borderRadius: 10,
    marginVertical: 20,
    shadowColor: "#000",
    shadowOffset: {
	    width: 0,
	    height: 2,
    },
    shadowOpacity: 0.23,
    shadowRadius: 2.62,
  },
  image: {
    width: 100,
    height: 100,
    justifyContent: 'center',
    alignItems: 'center',
  },
  box: {
    justifyContent: "center",
    margin: 10,
  },
  icon: {
    width: 20,
    height: 20,
  },
  baseText: {
    padding: 20,
    flexDirection: "column",
    justifyContent: "space-between"
  },
  titleText: {
    fontFamily: "nunito-bold",
    fontSize: 20,
  },
  descriptionText: {
    fontFamily: "nunito-regular",
    fontSize: 13,
  },
});
