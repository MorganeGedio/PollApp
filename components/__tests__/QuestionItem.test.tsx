import React from "react";
import renderer from "react-test-renderer";
import QuestionItem, { QuestionItemProps } from "../QuestionItem";

describe("QuestionItem", () => {
  const testProps = {
    title: "My question",
    date: "2020-05-25",
    navigateToQuestion: jest.fn(),
  };

  test("component renders correctly", () => {
    // render a component with the new renderer
    const component = renderer.create(<QuestionItem {...testProps} />);
    // transform it into JSON
    let tree = component.toJSON();
    // match the snapshot to the previously stored snapshot
    expect(tree).toMatchSnapshot();
  });

  
});
