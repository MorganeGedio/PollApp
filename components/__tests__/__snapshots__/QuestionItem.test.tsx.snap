// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionItem component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#dfcdc3",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 130,
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 20,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 2.62,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexGrow": 1,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "nunito-bold",
            "fontSize": 20,
            "marginBottom": 5,
          }
        }
      >
        My question
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "nunito-regular",
            "fontSize": 13,
            "marginBottom": 5,
          }
        }
      >
        Created on: 
        2020-05-25
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
      />
    </View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "height": 100,
            "width": 100,
          }
        }
      />
    </View>
  </View>
</View>
`;
